py-polars/polars/series/series.py

```diff
@@ -3568,9 +3582,20 @@ def search_sorted(
         ]
         """
         df = F.select(F.lit(self).search_sorted(element, side))
-        if isinstance(element, (list, Series, pl.Expr, np.ndarray)):
+        # These types unambiguously return a Series:
+        #
+        # * Series means we want to search for multiple values.
+        # * Expr because that always returns a Series, matching Expr.search_sorted().
+        if isinstance(element, (Series, pl.Expr)):
+            return df.to_series()
+
+        if (
+            isinstance(element, list)
+            or (_check_for_numpy(element) and isinstance(element, np.ndarray))
+        ) and not isinstance(self.dtype, (List, Array)):
             return df.to_series()
         else:
+            assert len(df) == 1
             return df.item()
```

Comment:
Not your fault or doing, but we are going to remove this at 2.0.

Reply:
Remove the "returning a single item" or "remove len()"? I.e. what change do you want me to do here?

Reply:
The fact that the eager version will return a single item here sometimes. I don't think we change anything for now. Maybe just add a comment `// @TAG: 2.0`.